arkouda/dataframe.py

```diff
@@ -1845,6 +1848,261 @@ def numeric_help(d):
+            This is an in-place modification, the original is returned to support a
+            fluid programming style.
+            Please note you cannot register two different DataFrames with the same name.
+
+        Raises
+        ------
+        TypeError
+            Raised if user_defined_name is not a str
+        RegistrationError
+            If the server was unable to register the DataFrame with the user_defined_name
+
+        See also
+        --------
+        unregister, attach, unregister_dataframe_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        self.index.register(f"df_index_{user_defined_name}")
+        array(self.columns).register(f"df_columns_{user_defined_name}")
+
+        for col, data in self.data.items():
+            data.register(f"df_data_{col}_{data.objtype}_{user_defined_name}")
+
+        self.name = user_defined_name
+        return self
+
+    def unregister(self):
+        """
+        Unregister this DataFrame object in the arkouda server which was previously
+        registered using register() and/or attached to using attach()
+
+        Raises
+        ------
+        RegistrationError
+            If the object is already unregistered or if there is a server error
+            when attempting to unregister
+
+        See also
+        --------
+        register, attach, unregister_dataframe_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if not self.name:
+            raise RegistrationError(
+                "This DataFrame does not have a name and does not appear to be registered."
+            )
+
+        DataFrame.unregister_dataframe_by_name(self.name)
+
+        self.name = None  # Clear our internal DataFrame object name
+
+    def is_registered(self) -> bool:
+        """
+        Return True if the object is contained in the registry
+
+        Returns
+        -------
+        bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mismatch of registered components
+
+        See Also
+        --------
+        register, attach, unregister, unregister_dataframe_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if self.name is None:
+            return False
+
+        # Total number of registered parts should equal number of columns plus an entry
+        # for the index and the column order
+        total = len(self.data) + 2
+        registered = 0
+
+        for col, data in self.data.items():
+            if data.is_registered():
+                registered += 1
```

Comment:
this way is totally fine but i wanted to throw out this alternative
```python
registered = sum([data.is_registered() for col, data in self.data.items()])
```

Reply:
Updated